File: src/java/foundgine-core/src/main/java/com/foundgine/core/semantic/metadata/AliasDeclaration.java
package com.foundgine.core.semantic.metadata;
/** Static declaration of a historical or synonymous semantic name. */
public record AliasDeclaration(String name, Integer weight) {
    public AliasDeclaration {
        if (name == null || name.isBlank()) throw new IllegalArgumentException("Alias name is required.");
        if (weight != null && (weight < 1 || weight > 100))
            throw new IllegalArgumentException("Alias weight must be between 1 and 100 (inclusive) when specified.");
    }
    public AliasDeclaration(String name) { this(name, null); }
    @Override public String toString() { return name; }
}

File: src/csharp/Foundgine.Core/Semantic/Metadata/AliasDeclaration.cs
namespace Foundgine.Core.Semantic.Metadata;

/// <summary>
/// Static declaration of a historical/synonymous semantic name, as discovered
/// from metadata (attributes or hand-authored catalogs). Mirrors
/// <see cref="Foundgine.Core.Semantic.SemanticAlias"/> but lives in the
/// metadata layer so <c>Foundgine.Core.Semantic</c> carries no dependency on
/// the concrete metadata assembly.
/// </summary>
public sealed record AliasDeclaration
{
    /// <summary>The alias name.</summary>
    public string Name { get; }

    /// <summary>
    /// Optional evidence weight in the inclusive range 1-100. Null
    /// when the declaration carries no weight.
    /// </summary>
    public int? Weight { get; }

    public AliasDeclaration(string name, int? weight = null)
    {
        if (weight is < 1 or > 100)
        {
            throw new ArgumentOutOfRangeException(
                nameof(weight),
                weight,
                "Alias weight must be between 1 and 100 (inclusive) when specified.");
        }

        Name = name;
        Weight = weight;
    }

    public override string ToString() => Name;
}
